monitor-core-test/src/test/java/net/spals/appbuilder/monitor/core/TracerProviderTest.java

```diff
@@ -0,0 +1,54 @@
+package net.spals.appbuilder.monitor.core;
+
+import io.opentracing.NoopTracer;
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link TracerProvider}
+ *
+ * @author tkral
+ */
+public class TracerProviderTest {
+
+    @Test
+    public void testNoopDefault() {
+        final TracerPlugin tracerPlugin = mock(TracerPlugin.class);
+        when(tracerPlugin.createTracer(anyMap())).thenReturn(NoopTracerFactory.create());
+
+        final TracerProvider tracerProvider = new TracerProvider(
+                Collections.singletonMap("noop", tracerPlugin),
+                Collections.<String, TracerTag>emptyMap()
+        );
+
+        assertThat(tracerProvider.get(), instanceOf(NoopTracer.class));
+    }
+
+    // Run after testNoopDefault because only one GlobalTracer is allowed
+    @Test(dependsOnMethods = "testNoopDefault")
```

Comment:
Sweet. I haven't tried using these, I really should.

Reply:
That being dependency methods.

Reply:
I usually use them for integration tests because test A sets up some state for test B. Though, the test ordering can be a bit dangerous in that sense. Maven and IntelliJ both adhere to test dependency order, so that isn't a problem. But debugging can be a PIA sometimes. Almost as a rule, I try to avoid this in unit tests because those tests should stand independently on their own. This was a special case because we're having to deal with a static reference.

Reply:
I thought was for if you have unit tests then your ftest should make sure the unit tests pass to bother running.